yeonjae/programmers-1/최소직사각형.js

```diff
@@ -0,0 +1,30 @@
+function solution(sizes) {
+  const answer = [];
+
+  for (let i = 0; i < sizes.length; i++) {
+    if (sizes[i][0] < sizes[i][1]) {
+      answer.push(sizes[i].reverse());
+    } else {
+      answer.push(sizes[i]);
+    }
+  }
+
+  const width = [];
+  const height = [];
+
+  for (let i = 0; i < answer.length; i++) {
+    width.push(answer[i][0]);
+    height.push(answer[i][1]);
+  }
+  const sortWidth = width.sort((a, b) => b - a);
+  const sortHeight = height.sort((a, b) => b - a);
+
+  return sortWidth[0] * sortHeight[0];
+}
```

Comment:
```suggestion
function solution(sizes) {
  const sortedSizes = sizes.map(([width, height]) =>
    width < height ? [height, width] : [width, height]
  );
  const height = [];
  const width = [];

  for (let i = 0; i < sortedSizes.length; i++) {
    width.push(sortedSizes[i][0]);
    height.push(sortedSizes[i][1]);
  }

  return Math.max(...width) * Math.max(...height);
}
```
- 풀이 잘봤습니다 🙆🏻‍♂️ 
- 첫 번째 for문을 통해서 sizes의 각 요소를 내림차순 정렬을 해주신 것 같은데 map을 이용한 정렬이 좀 더 깔끔하다고 느꼈습니다.
- 또한, sortWidth와 sortHeight는 answer에 의존하고 있기 때문에 어쩔 수 없겠지만 width, height를 코드 상단으로 뺴주셔야됩니다. 그리고 변수명도 `정렬이 된! width` 라는 의미가 커서 sortedWidth가 적절해 보입니다.
- 위에 풀이는 불 필요한 로직을 제거하고, 좀 더 효율적으로 바꾼 코드입니다. 참고해주세요 
- 아래 코드는 배열이 아닌 숫자 변수를 재할당하는 것과 reduce를 활용한 방법입니다. 참고해주세요~ 🙇🏻‍♂️
```js
function solution(sizes) {
  const sortedSizes = sizes.map(([width, height]) =>
    width < height ? [height, width] : [width, height]
  );
  let [maxWidth, maxHeight] = [0, 0];

  return sortedSizes.reduce((_, cur) => {
    const [width, height] = cur;

    maxWidth = Math.max(width, maxWidth);
    maxHeight = Math.max(height, maxHeight);

    return maxWidth * maxHeight;
  }, 0);
}
```